tree/dataframe/inc/ROOT/RDFInterface.hxx

```diff
@@ -111,6 +111,7 @@ class RInterface {
    ColumnNames_t fValidCustomColumns; ///< Names of columns `Define`d for this branch of the functional graph.
    /// Non-owning pointer to a data-source object. Null if no data-source. RLoopManager has ownership of the object.
    RDataSource *const fDataSource = nullptr;
+   std::shared_ptr<const ColumnNames_t> fDatasetColumns; ///< Cache of the chain columns names
```

Comment:
If I understand correctly the variable should contain just the TTree columns, if any, not all the columns. And it's all the possible variations of valid column names, so each column is in there multiple times...maybe `fValidBranchNames`?

Reply:
That's a good point. I was not sure what name to use and if the name should avoid ttree jeargon. Since this is really about trees and is a private data member, we can probably use branch rather than column.